src/main/java/io/debezium/connector/yugabytedb/YugabyteDBChangeEventSourceCoordinator.java

```diff
@@ -173,6 +188,34 @@ public void commitOffset(Map<String, ?> offset) {
         }
     }
 
+    /**
+     * @return the set of partitions i.e. {@link YBPartition} being in the streaming phase at a
+     * given point in time. If streamingChangeEventSource is null that means we are still in the
+     * snapshot phase and in that case it should be safe to return an {@link Optional#empty()}
+     * which should be handled by the caller of this method.
+     */
+    public Optional<Set<YBPartition>> getPartitions() {
+        if (this.streamingChangeEventSource == null) {
+            LOGGER.info("Streaming change event source is null, returning empty value");
+            return Optional.empty();
+        }
+
+        Optional<Set<YBPartition>> ybPartitions =
+          Optional.of(this.streamingChangeEventSource.getActivePartitionsBeingPolled());
```

Comment:
this.streamingSource == null || (this.stteamingSource.getActivePartitionsbegingPolled().size() == 0)

- check if there's an empty() method for set.